pycbc/strain/strain.py

```diff
@@ -373,49 +263,133 @@ def from_cli(opt, dyn_range_fac=1, precision='single',
-
-        if opt.sgburst_injection_file:
-            logging.info("Applying sine-Gaussian burst injections")
-            injector =  SGBurstInjectionSet(opt.sgburst_injection_file)
-            injector.apply(strain, opt.channel_name[0:2],
-                             distance_scale=opt.injection_scale_factor)
-
-        if opt.strain_high_pass:
-            logging.info("Highpass Filtering")
-            strain = highpass(strain, frequency=opt.strain_high_pass)
-
+            strain = colored_noise(strain_psd,
+                                   opt.gps_start_time - opt.pad_data,
+                                   opt.gps_end_time + opt.pad_data,
+                                   seed=opt.fake_strain_seed,
+                                   low_frequency_cutoff=lowfreq)
+
+    if not opt.channel_name and (opt.injection_file \
+                                 or opt.sgburst_injection_file):
+        raise ValueError('Please provide channel names with the format '
+                         'ifo:channel (e.g. H1:CALIB-STRAIN) to inject '
+                         'simulated signals into fake strain')
+
+    if opt.zpk_z and opt.zpk_p and opt.zpk_k:
+        logging.info("Highpass Filtering")
+        strain = highpass(strain, frequency=opt.strain_high_pass)
+
+        logging.info("Applying zpk filter")
+        z = numpy.array(opt.zpk_z)
+        p = numpy.array(opt.zpk_p)
+        k = float(opt.zpk_k)
+        strain = filter_zpk(strain.astype(numpy.float64), z, p, k)
+
+    if opt.normalize_strain:
+        logging.info("Dividing strain by constant")
+        l = opt.normalize_strain
+        strain = strain / l
+
+    if opt.strain_high_pass:
+        logging.info("Highpass Filtering")
+        strain = highpass(strain, frequency=opt.strain_high_pass)
+
+    if opt.sample_rate:
         logging.info("Resampling data")
-        strain = resample_to_delta_t(strain, 1. / opt.sample_rate)
-
-        if precision == 'single':
-            logging.info("Converting to float32")
-            strain = (dyn_range_fac * strain).astype(pycbc.types.float32)
-        elif precision == 'double':
-            logging.info("Converting to float64")
-            strain = (dyn_range_fac * strain).astype(pycbc.types.float64)
-        else:
-            raise ValueError("Unrecognized precision {}".format(precision))
+        strain = resample_to_delta_t(strain,
+                                     1. / opt.sample_rate,
+                                     method='ldas')
 
-        if opt.strain_high_pass:
-            logging.info("Highpass Filtering")
-            strain = highpass(strain, frequency=opt.strain_high_pass)
+    if injector is not None:
+        logging.info("Applying injections")
+        injections = \
+            injector.apply(strain, opt.channel_name[0:2],
+                           distance_scale=opt.injection_scale_factor,
+                           injection_sample_rate=opt.injection_sample_rate,
```

Comment:
Okay, good. Injections now made after resampling. Will have to make sure to update ini files for running with TD injections though (at least EOB will probably fail if we forget to do this).